### PR TITLE
Bugfix calls

### PR DESCRIPTION
**Pull request by norhh**

### CODE REVIEW

**Comment by b-mueller** [comment]
@norhh it still doesn't detect this assert violation:

```
pragma solidity ^0.5.8;
pragma experimental ABIEncoderV2;

interface IERC20
{
	function totalSupply()
		external view returns (uint256);

	function balanceOf(address who)
		external view returns (uint256);

	function allowance(address owner, address spender)
		external view returns (uint256);

	function transfer(address to, uint256 value)
		external returns (bool);

	function approve(address spender, uint256 value)
		external returns (bool);

	function transferFrom(address from, address to, uint256 value)
		external returns (bool);

	event Transfer(
		address indexed from,
		address indexed to,
		uint256 value
	);

	event Approval(
		address indexed owner,
		address indexed spender,
		uint256 value
	);
}

contract SomeAssert
{

	IERC20 public token;

	constructor(address _token)
	public
	{
		token = IERC20(0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa);
	}

	function run(uint256 _amount)
	public returns (bool)
	{
		require(token.transferFrom(msg.sender, address(this), _amount));
		assert(false);
	}
}
```

I tried:

```
$ $MYTH analyze someassert.sol:SomeAssert -t1 -mexceptions
The analysis was completed successfully. No issues were detected.

```

File: mythril/laser/ethereum/instructions.py [hunk 1]
```diff
@@ -1837,6 +1833,20 @@ def stop_(self, global_state: GlobalState):
         """
         global_state.current_transaction.end(global_state)
 
+    @staticmethod
+    def _write_symbolic_memory(
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: Can you add a docstring to this function?

Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: What do you think about `_write_symbolic_return_data`, write_symbolic_memory seems fairly generic

File: mythril/analysis/modules/external_calls.py [hunk 2]
```diff
@@ -91,17 +91,16 @@ def _analyze_state(self, state: GlobalState):
                 state, constraints + state.world_state.constraints
             )
 
+            for tx in state.world_state.transaction_sequence:
+                if not isinstance(tx, ContractCreationTransaction):
+                    constraints.append(tx.caller == ACTORS.attacker)
             # Check whether we can also set the callee address
```
Comment (JoranHonig) [inline mythril/analysis/modules/external_calls.py]: What is the reason for the changes in this file?

**Comment by norhh** [comment]
@b-mueller can you increase the max-depth to a bigger number, max-depth 50 is apparently too low for that.

File: mythril/analysis/modules/external_calls.py (hunk 2 above)
Comment (norhh) [reply mythril/analysis/modules/external_calls.py]: The contract provided pointed to two bugs, one is in calls and the other resulted in a buggy output after I fixed the call one

**Comment by b-mueller** [comment]
Can we also change `max_depth` to 128 with this PR? The original main intention of max_depth was to skip loops which shouldn't be an issue now.

**Comment by b-mueller** [comment]
Is this ready for merging?

File: mythril/laser/ethereum/instructions.py [hunk 3]
```diff
@@ -1837,6 +1833,27 @@ def stop_(self, global_state: GlobalState):
         """
         global_state.current_transaction.end(global_state)
 
+    @staticmethod
+    def _write_symbolic_calldata(
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: The calldata is something different from the return data right?
In this case you seem to be returning the latter

File: mythril/analysis/modules/external_calls.py [hunk 4]
```diff
@@ -91,17 +91,16 @@ def _analyze_state(self, state: GlobalState):
                 state, constraints + state.world_state.constraints
             )
 
+            for tx in state.world_state.transaction_sequence:
+                if not isinstance(tx, ContractCreationTransaction):
+                    constraints.append(tx.caller == ACTORS.attacker)
```
Comment (JoranHonig) [inline mythril/analysis/modules/external_calls.py]: Should this not be something that appends to new_constraints?

Comment (JoranHonig) [reply mythril/analysis/modules/external_calls.py]: I don't think it should be part of the constraints reported if the next sat soliving fails
